integration/test/metrics_number_dimension/metrics_number_dimension_test.go

```diff
@@ -0,0 +1,133 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: MIT
+
+//go:build linux && integration
+// +build linux,integration
+
+package metrics_number_dimension
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws/amazon-cloudwatch-agent/integration/test"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch"
+	"github.com/aws/aws-sdk-go-v2/service/cloudwatch/types"
+	"log"
+	"strings"
+	"testing"
+	"time"
+)
+
+const configOutputPath = "/opt/aws/amazon-cloudwatch-agent/bin/config.json"
+const configJSON = "/config.json"
+const namespace = "MetricNumberDimensionTest"
+const instanceId = "InstanceId"
+const append = "append"
+
+//Let the agent run for 2 minutes. This will give agent enough time to call server
+const agentRuntime = 2 * time.Minute
+
+var targetString = "max MaxDimensions %v is less than than number of dimensions %v thus only taking the max number"
+
+type input struct {
+	dataInput            string
+	findTarget           bool
+	inputNumberDimension int
+	metricName           string
+}
+
+type metric struct {
+	name  string
+	value string
+}
+
+//Must run this test with parallel 1 since this will fail if more than one test is running at the same time
+//This test uses a pem file created for the local stack endpoint to be able to connect via ssl
+func TestBundle(t *testing.T) {
+
+	parameters := []input{
+		{dataInput: "resources/10_dimension",
+			findTarget:           false,
+			inputNumberDimension: 10,
+			metricName:           "mem_used_percent"},
+		// @TODO add when https://github.com/aws/amazon-cloudwatch-agent/pull/361 is merged
+		// {dataInput: "resources/30_dimension", findTarget: false, inputNumberDimension: 30, metricName: "mem_used_percent"},
```

Comment:
If we were to turn on this test today, would this have `findTarget = true` to pass the test? If that's the case, I think that what we can do is
1. uncomment these test cases so we can merge this fully
2. modify the other PR to account for this by flipping the bool here, so we can illustrate that our integration tests do what we want them to do 😄

Reply:
It will not find true since the other pr is not merged. The log target is added in that pr.